### PR TITLE
Fixed windows MSVC build compatibility

### PR DESCRIPTION
**Pull request by yuanluxu**
Fixed a few MSVC compiler (visual studio 2019, MSVC 19.16.27034) compatibility issues
1. Replaced long with int64_t. aten::data_ptr\<long\> is not supported in MSVC
2. pytorch3d/csrc/rasterize_points/rasterize_points_cpu.cpp, inline function is not correctly recognized by MSVC. 
3. pytorch3d/csrc/rasterize_meshes/geometry_utils.cuh
const auto kEpsilon = 1e-30;
MSVC does not compile this const into both host and device, change to a MACRO. 
4. pytorch3d/csrc/rasterize_meshes/geometry_utils.cuh, 
const float area2 = pow(area, 2.0);
2.0 is considered as double by MSVC and raised an error
5. pytorch3d/csrc/rasterize_points/rasterize_points_cpu.cpp
std::tuple<torch::Tensor, torch::Tensor> RasterizePointsCoarseCpu() return type does not match the declaration in rasterize_points_cpu.h.

### CODE REVIEW

File: pytorch3d/csrc/nearest_neighbor_points/nearest_neighbor_points.cu [hunk 1]
```diff
@@ -219,7 +219,7 @@ at::Tensor nn_points_idx_cuda(at::Tensor p1, at::Tensor p2) {
   const auto D = p1.size(2);
 
   AT_ASSERTM(p2.size(2) == D, "Point sets must have same last dimension.");
-  auto idx = at::empty({N, P1}, p1.options().dtype(at::kLong));
+  auto idx = at::empty({N, P1}, p1.options().dtype(at::kint64_t));
```
Comment (nikhilaravi) [inline pytorch3d/csrc/nearest_neighbor_points/nearest_neighbor_points.cu]: The Circle CI build is failing as `kint64_t`  is not a valid torch variable. Was `at::kLong` causing an issue?

see: https://pytorch.org/cppdocs/api/variable_namespacetorch_1aca5be74e0c90ee0b02abcdf224231c30.html?highlight=klong

Comment (yuanluxu) [reply pytorch3d/csrc/nearest_neighbor_points/nearest_neighbor_points.cu]: Ah. Sorry this should be kept as it is.

**Comment by nikhilaravi** [comment]
@yuanluxu Apologies for the delay in responding to this PR. We're working on windows support so stay tuned!

**Comment by bottler** [comment]
Can I clarify the two preprocessor definitions? 

I think `const float kEpsilon = 1e-30f;` would be fine for all compilers.

I think the problem for `PixToNdc` is that the same function appears in two `cpu.cpp` files. Can we just change `inline float PixToNdc` to `static float PixToNdx`, or put it in an anonymous namespace, in one or both places?

Thanks

**Comment by yuanluxu** [comment]
@bottler 
1. For `const float kEpsilon = 1e-30f;`, I actually tried this one and several other variants and you would get the same compliant from msvc (see below). This might be an issue for msvc. msvc does not create two copies of the constant on both host and device while gcc does by default. 

c:\pytorch3d\pytorch3d\csrc\rasterize_meshes\geometry_utils.cuh(236): error: identifier "kEpsilon" is undefined in device code

C:/pytorch3d/pytorch3d/csrc/rasterize_meshes/rasterize_meshes.cu(390): error: identifier "kEpsilon" is undefined in device code

C:/pytorch3d/pytorch3d/csrc/rasterize_meshes/rasterize_meshes.cu(426): error: identifier "kEpsilon" is undefined in device code

C:/pytorch3d/pytorch3d/csrc/rasterize_meshes/rasterize_meshes.cu(133): error: identifier "kEpsilon" is undefined in device code

C:/pytorch3d/pytorch3d/csrc/rasterize_meshes/rasterize_meshes.cu(133): error: identifier "kEpsilon" is undefined in device code

C:/pytorch3d/pytorch3d/csrc/rasterize_meshes/rasterize_meshes.cu(141): error: identifier "kEpsilon" is undefined in device code

6 errors detected in the compilation of "C:/Users/yuanluxu/AppData/Local/Temp/tmpxft_000113dc_00000000-10_rasterize_meshes.cpp1.ii".
error: command 'C:\\Program Files\\NVIDIA GPU Computing Toolkit\\CUDA\\v10.2\\bin\\nvcc.exe' failed with exit status 1

2. Changing from `inline float PixToNdc` to `static float PixToNdx` works! I updated my fork!

**Comment by bottler** [comment]
Thanks! Looks good to me.

**Comment by nikhilaravi** [comment]
@yuanluxu it seems there are conflicts with master now (some files have been updated recently). Can you fix these conflicts and update the PR? Thank you!

**Comment by yuanluxu** [comment]
@nikhilaravi Fixed!

**Comment by nikhilaravi** [comment]
Looks good! Thanks @yuanluxu for adding this!

**Pull request closed without merging**